lib/core/test/integration/Test/Integration/Scenario/CLI/Mnemonics.hs

```diff
@@ -34,10 +34,68 @@ spec = do
         v `shouldBe` (showVersion version)
         c `shouldBe` ExitSuccess
 
-    it "CLI_HELP - cardano-wallet shows help on bad argument" $  do
-        (Exit c, Stdout out) <- cardanoWalletCLI ["--bad arg"]
-        out `shouldContain` "Cardano Wallet CLI"
-        c `shouldBe` ExitFailure 1
+    describe "CLI_HELP - cardano-wallet shows help on bad arg or param" $ do
+        let wid = replicate 40 '1'
+        let badArgs =
+                [ "--bad param"
+                , "bad arg"
+                -- serve
+                , "serve --"
+                , "serve --network testnet --random-por --quiet"
+                , "serve --network testnet --random-port --bridge-port 666 --vebrose"
+                , "serve --network"
+                , "server --verbose"
+                , "serve --database"
+                --launch
+                , "launch --quit"
+                , "launch --network mainnet --state-dir"
+                , "launch --randomport 666 --state-dir tempdir"
+                , "launcher --verbose"
+                --mnemonic
+                , "mnemonic generate --size"
+                , "mnemnic generate"
+                , "mnemonic --size 15"
+                -- wallet list
+                , "wallet list --port"
+                -- wallet create
+                , "wallet create"
+                , "wallet"
+                , "create"
+                , "wallet crate name"
+                , "wallet create --port name"
+                , "wallet create --address-pool-gap 30 name"
+                , "wallet create name --address-pool-gap"
+                -- wallet get
+                , "wallet get"
+                , "get"
+                , "get " ++ wid
+                , "wallet get " ++ wid ++ " --port"
+                -- wallet update
+                , "wallet update"
+                , "update"
+                , "update " ++ wid
+                , "wallet update " ++ wid ++ " --port"
+                -- wallet delete
+                , "wallet delete"
+                , "delete"
+                , "delete " ++ wid
+                , "wallet delete " ++ wid ++ " --port"
+                -- transaction create
+                , "transaction create " ++ wid ++ " --payment"
+                , "transaction create " ++ wid ++ " --payment 22@2cWKMJemoBaiPcjZZKQzTHzjRkaAee5dx246Ren8U5KcDGt9QX6FZQPskzykhYL1AW62U --payment"
+                , "transaction create --port " ++ wid ++ " --payment 22@ --payment"
```

Comment:
I was expecting to reach [this](https://github.com/input-output-hk/cardano-wallet/blob/master/lib/cli/src/Cardano/CLI.hs#L231) when trying out different variations on `--payment` parameter but was never able to...

(perhaps this is never reached?)

Reply:
This seems to be caught earlier by the parser itself (docopt) and never makes it to our code.